compiler/fir/checkers/src/org/jetbrains/kotlin/fir/analysis/checkers/declaration/FirTypeAliasChecker.kt

```diff
@@ -0,0 +1,25 @@
+/*
+ * Copyright 2010-2021 JetBrains s.r.o. and Kotlin Programming Language contributors.
+ * Use of this source code is governed by the Apache 2.0 license that can be found in the license/LICENSE.txt file.
+ */
+
+package org.jetbrains.kotlin.fir.analysis.checkers.declaration
+
+import org.jetbrains.kotlin.fir.analysis.checkers.context.CheckerContext
+import org.jetbrains.kotlin.fir.analysis.diagnostics.DiagnosticReporter
+import org.jetbrains.kotlin.fir.analysis.diagnostics.FirErrors
+import org.jetbrains.kotlin.fir.analysis.diagnostics.reportOn
+import org.jetbrains.kotlin.fir.declarations.FirDeclaration
+import org.jetbrains.kotlin.fir.declarations.FirFile
+import org.jetbrains.kotlin.fir.declarations.FirTypeAlias
+
+object FirTypeAliasChecker : FirBasicDeclarationChecker() {
+    override fun check(declaration: FirDeclaration, context: CheckerContext, reporter: DiagnosticReporter) {
+        if (declaration !is FirTypeAlias) return
+        if (context.containingDeclarations.lastOrNull() !is FirFile) {
+            reporter.reportOn(declaration.source, FirErrors.TOPLEVEL_TYPEALIASES_ONLY, context)
+        }
+
+        FirExposedVisibilityDeclarationChecker.checkTypeAlias(declaration, reporter, context)
```

Comment:
Hm, is it just me or does this look awaked that `FirTypeAliasChecker` is calling `checkTypeAlias` in a different checker? :) Maybe in a separate commit, how about migrating that one to here and call that from `FirExposedVisibilityDeclarationChecker`?

Also, I think adding this doesn't change any test data, which means... we don't test whether top-level type alias is exposing other types?

Reply:
Good point! Let me add a test case for that.